CMakeLists.txt

```diff
@@ -26,11 +26,15 @@ endif()
 
 find_package(Threads REQUIRED)
 find_package(Python3 3.6 REQUIRED)
-if(NOT DISABLE_MPI)
-    find_package(MPI REQUIRED)
+if(DEFINED USE_MPI)
+    if(USE_MPI)
+        find_package(MPI REQUIRED)
+    endif()
+else()
+    find_package(MPI)
 endif()
```

Comment:
I am not entirely sure if it is a good idea to say that we don't have MPI by default. Why are you proposing this change?

Reply:
Maybe the change is not clear. The objective is the following:
- If we invoke CMake without specifying a value for ```USE_MPI```, then we will try to find MPI: if we find it, we build with it, otherwise we build without
- If specify ```USE_MPI=ON``` then we will look for MPI and if we don't find it, we error out
- If specify ```USE_MPI=OFF``` then we build without MPI